src/components/pages/devices/flyouts/deviceDelete/deviceDelete.js

```diff
@@ -0,0 +1,135 @@
+  Btn,
+  BtnToolbar,
+  ErrorMsg,
+  Flyout,
+  FlyoutHeader,
+  FlyoutTitle,
+  FlyoutCloseBtn,
+  FlyoutContent,
+  Indicator
+} from 'components/shared';
+
+import './deviceDelete.css';
+
+export class DeviceDelete extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      physicalDevices: [],
+      containsSimulatedDevices: false,
+      isPending: false,
+      error: undefined,
+      successCount: 0,
+      changesApplied: false
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.devices) {
+      this.populateDevicesState(this.props.devices);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.devices && (this.props.devices || []).length !== nextProps.devices.length) {
+      this.populateDevicesState(nextProps.devices);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.subscription) this.subscription.unsubscribe();
+  }
+
+  populateDevicesState = (devices) => {
+    const deviceArr = (devices || []);
+    const physDevices = deviceArr.filter(function (d) { return !d.isSimulated });
+    this.setState({ physicalDevices: physDevices, containsSimulatedDevices: (physDevices.length !== deviceArr.length) });
+  }
+
+  deleteDevices = () => {
+    this.setState({ isPending: true });
+
+    this.subscription = Observable.from(this.state.physicalDevices)
+      .flatMap(({ id }) =>
+        IoTHubManagerService.deleteDevice(id)
+          .map(() => id)
+      )
+      .subscribe(
+        next => {
+          this.setState({ successCount: this.state.successCount + 1 });
+          return this.props.deleteDevice(next);
+        },
+        error => this.setState({ error, isPending: false, changesApplied: true }),
+        () => this.setState({ isPending: false, changesApplied: true })
+      );
+
+  }
+
+  render() {
+    const { t, onClose } = this.props;
+    const { physicalDevices, containsSimulatedDevices, isPending, error, successCount, changesApplied } = this.state;
+
+    const summaryCount = changesApplied ? successCount : physicalDevices.length;
+    const summaryMessage = changesApplied ? t('devices.delete.applySuccess') : t('devices.delete.affected');
+
+    return (
+      <Flyout>
+        <FlyoutHeader>
+          <FlyoutTitle>{t('devices.delete.title')}</FlyoutTitle>
+          <FlyoutCloseBtn onClick={onClose} />
+        </FlyoutHeader>
+        <FlyoutContent>
+          <div className="device-delete-container">
+            <div className="device-delete-header">{t('devices.delete.header')}</div>
+            <div className="device-delete-descr">{t('devices.delete.description')}</div>
+            {
+              containsSimulatedDevices &&
+              <div className="simulated-device-selected">
+                <Svg path={svgs.infoBubble} className="info-icon" />
+                <div>{t('devices.delete.simulatedNotSupported')}</div>
+              </div>
+            }
+            <div className="device-delete-summary-container">
+              <div>
```

Comment:
empty div

Reply:
This changes to a "summarySection" in my next PR... so will leave as is for now.